internal/osbuilconfig/osbuilconfig_handler.go

```diff
@@ -0,0 +1,48 @@
+package osbuilconfig
+
+import (
+	"context"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/project-flotta/osbuild-operator/internal/maniffests"
+	"github.com/project-flotta/osbuild-operator/internal/repository/osbuild"
+	"github.com/project-flotta/osbuild-operator/internal/repository/osbuildconfig"
+	"github.com/project-flotta/osbuild-operator/restapi/operations/osbuilconfig"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type OSBuildConfigHandler struct {
+	OSBuildConfigRepository osbuildconfig.Repository
+	OSBuildRepository       osbuild.Repository
+	Scheme                  *runtime.Scheme
+}
+
+func NewOSBuildConfigHandler(osBuildConfigRepository osbuildconfig.Repository, osBuildRepository osbuild.Repository, scheme *runtime.Scheme) *OSBuildConfigHandler {
+	return &OSBuildConfigHandler{
+		OSBuildConfigRepository: osBuildConfigRepository,
+		OSBuildRepository:       osBuildRepository,
+		Scheme:                  scheme,
+	}
+}
+
+func (o *OSBuildConfigHandler) OSBuildConfigWebhookTriggers(ctx context.Context, params osbuilconfig.OSBuildConfigWebhookTriggersParams) middleware.Responder {
+	logger := log.FromContext(ctx)
+	logger.Info("OSBuildConfigWebhookTriggerHandler new request")
+
+	osBuildConfig, err := o.OSBuildConfigRepository.Read(ctx, params.OsbuildconfigCrName, params.OsbuildconfigNamespaceName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return osbuilconfig.NewOSBuildConfigWebhookTriggersNotFound()
+		}
+		logger.Error(err, "can't read OSBuildConfigCR")
+		return osbuilconfig.NewOSBuildConfigWebhookTriggersBadRequest()
```

Comment:
Why is this a BadRequest and not an InternalServerError?

Reply:
I wasn't sure here. 
If you think it should be internalServerError I will change it